src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,240 @@
+                shape=PosixPath('/home/davidhaasnoot/eWaterCycle-WSL-WIP/Forcing/Camels/shapefiles/camels_01022500.shp'),
+                filenames={
+                    'tasmax': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmax.nc',
+                    'tasmin': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmin.nc',
+                    'evspsblpot': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_evspsblpot.nc',
+                    'pr': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_pr.nc',
+                    'tas': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tas.nc',
+                    'Q': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_Q.nc'
+                }
+            )
+    """
+
+    @classmethod
+    def retrieve(cls: Type["Caravan"],
+                 start_time: str,
+                 end_time: str,
+                 basin_id: str,
+                 directory: str | Path,
+                 variables: tuple[str, ...] = (),
+                 shape: str | Path | None = None,
+                 **kwargs) -> "Caravan":
+        """Retrieve caravan for a model.
+
+        Args:
+            start_time: Start time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            end_time: nd time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            basin_id: Caravan basin_id
+            directory: Directory in which forcing should be written.
+                If not given will create timestamped directory.
+            variables: Variables which are needed for model,
+                if not specified will default to all.
+            shape: (Optional) Path to a shape file.
+                If none is specified, will be downloaded automatically.
+            **kwargs
+        """
+        dataset = basin_id.split('_')[0]
+        ds = xr.open_dataset(
+            f"{OPENDAP_URL}{dataset}.nc")
+        ds_basin = ds.sel(basin_id=basin_id.encode())
+        ds_basin_time = crop_ds(ds_basin, start_time, end_time)
+
+        if shape is None:
+            shape = get_shapefiles(Path(directory), basin_id)
+
+        if 'lat' not in variables and 'lon' not in variables:
+            shape_obj = shapereader.Reader(shape)
+            centre = next(shape_obj.geometries()).centroid
+            ds_basin_time.coords.update({'lat': centre.y,
+                                         'lon': centre.x
+                                         })
+        if variables == ():
+            variables = (ds_basin_time.data_vars.keys())
+
+        # only return the properties which are also in property vars
+        properties = tuple(set(variables).intersection(set(PROPERTY_VARS)))
+        variables = tuple(
+            set(variables).difference(set(PROPERTY_VARS)).intersection(
+              set(RENAME_ERA5.keys())
+            )) # only take the vars also in Rename dict
```

Comment:
I think this should do the same, but is more readable:
```suggestion
        non_property_vars = set(variables) - properties
        variable_names= non_propery_vars.intersection(RENAME_ERA5.keys())
```